multimap/src/main/java/org/infinispan/multimap/impl/HashMapBucket.java

```diff
@@ -0,0 +1,130 @@
+package org.infinispan.multimap.impl;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.infinispan.commons.marshall.ProtoStreamTypeIds;
+import org.infinispan.marshall.protostream.impl.MarshallableUserObject;
+import org.infinispan.multimap.internal.MultimapDataConverter;
+import org.infinispan.protostream.annotations.ProtoFactory;
+import org.infinispan.protostream.annotations.ProtoField;
+import org.infinispan.protostream.annotations.ProtoTypeId;
+
+@ProtoTypeId(ProtoStreamTypeIds.MULTIMAP_HASH_MAP_BUCKET)
+public class HashMapBucket<K, V> {
+
+   final Map<Object, Object> values;
+
+   // We don't transfer the converter in the proto message.
+   private MultimapDataConverter<K, V> converter = null;
+
+   public HashMapBucket() {
+      this.values = Collections.emptyMap();
+   }
+
+   public HashMapBucket(MultimapDataConverter<K, V> converter, K key, V value) {
+      this.values = new HashMap<>();
+      values.put(key, value);
+      this.converter = converter;
+   }
+
+   private HashMapBucket(MultimapDataConverter<K, V> converter, Map<K, V> values) {
+      this.converter = converter;
+      this.values = toStore(values);
+   }
+
+   @ProtoFactory
+   HashMapBucket(Collection<BucketEntry<Object, Object>> wrappedValues) {
+      this.values = wrappedValues.stream().collect(Collectors.toMap(Function.identity(), Function.identity()));
+      this.converter = null;
+   }
+
+   public static <K, V> HashMapBucket<K, V> create(MultimapDataConverter<K, V> converter, Map<K, V> values) {
+      return new HashMapBucket<>(converter, values);
+   }
+
+   @ProtoField(number = 1, collectionImplementation = ArrayList.class)
+   Collection<BucketEntry<Object, Object>> getWrappedValues() {
+      return values.entrySet().stream().map(BucketEntry::new).collect(Collectors.toList());
+   }
+
+   public Map<K, V> putAll(Map<K, V> map) {
```

Comment:
Any reason this method is returning the previous values? It seems we only care about the number of entries that were updated. If we only return that we also don't have to worry about doing any conversion fromStore either.

Reply:
Not sure why, I thought the `Map#putAll` returned the previous entries. 

Updated this to only return the # of created entries.